fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledRetryingIterator.java

```diff
@@ -538,11 +541,27 @@ public Builder<T> withNumOfRetries(int numOfRetries) {
             return this;
         }
 
+        /**
+         * Set the MDC context for the runner/executor.
+         * This MDC context will be carried out into the runner and executor and will allow them to pass that down to
+         * LOGGER calls used by the item handlers.
+         * Defaults to empty context.
+         * @param mdcContext the MDC context to use
+         * @return this builder
+         */
+        public Builder<T> withMdcContext(Map<String, String> mdcContext) {
+            this.contextConfigBuilder.setMdcContext(mdcContext);
+            return this;
+        }
+
         /**
          * Create the iterator.
          * @return the newly minted iterator
          */
         public ThrottledRetryingIterator<T> build() {
+            this.transactionalRunner = new TransactionalRunner(database, contextConfigBuilder);
```

Comment:
If the constructor that takes a `TransactionalRunner` is called, the one provided will not be used here.
That call does not appear to be used, maybe worth just deleting.